### PR TITLE
Update the `cfg` docs.

### PR DESCRIPTION
**Pull request by vpetrovykh**

### CODE REVIEW

File: docs/stdlib/cfg.rst [hunk 1]
```diff
@@ -214,14 +228,109 @@ Client connections
...
+
+  Every extension is expected to define its own extension-specific config
+  object type extending ``cfg::ExtensionConfig``. Any necessary extension
+  configuration setting should be represented as properties of this concrete
+  config type.
+
+  Up to three instances of the extension-specific config type will be created,
+  each of them with a ``required single link cfg`` to one of the
+  :eql:type:`cfg::Config`, :eql:type:`cfg::DatabaseConfig`, or
+  :eql:type:`cfg::InstanceConfig` according to the configuration level. The
```
Comment (raddevon) [inline docs/stdlib/cfg.rst:301]:
```suggestion
  each of them with a ``required single link cfg`` to the
  :eql:type:`cfg::Config`, :eql:type:`cfg::DatabaseConfig`, or
  :eql:type:`cfg::InstanceConfig` object depending on to the configuration
  level. The
```

**Review by raddevon — APPROVED** [review]
👏

File: docs/stdlib/cfg.rst [hunk 2]
```diff
@@ -214,14 +228,109 @@ Client connections
 ----------
 
 
+.. eql:type:: cfg::Config
+
+  The main configuration object type.
+
+  This type is supposed to have exactly one object instance. The
```
[the remark this answers is not in the document]
Comment (vpetrovykh) [reply docs/stdlib/cfg.rst:235]: Mainly the implication is that when you query this you should expect just one result. I wouldn't have even mentioned it if it weren't for `ExtensionConfig` having several objects that can get confusing if you aren't careful with a filter or don't access them via `extensions` link.

[the remark this answers is not in the document]
Comment (raddevon) [reply docs/stdlib/cfg.rst:235]: Ah, OK. Would something like "This type will have only one object instance," be accurate? Something about "is supposed to" here suggests to me an imminent failure state the user needs to diligently watch for.

**Pull request merged**